firefox_ui_tests/functional/locationbar/test_suggest_bookmarks.py

```diff
@@ -74,15 +74,18 @@ def visit_urls():
 
         # Wait for the search string to be present, for the autocomplete results to appear
         # and for there to be exactly one autocomplete result
-        self.wait_for_condition(lambda mn: locationbar.value == search_string)
-        self.wait_for_condition(lambda mn: autocomplete_results.is_open)
-        self.wait_for_condition(lambda mn: len(autocomplete_results.visible_results) == 2)
+        Wait(self.marionette).until(
+          lambda mn: (
+            locationbar.value == search_string and autocomplete_results.is_open
+          )
```

Comment:
I would suggest that we split this up to two Wait.until() calls. Having a combined check like that will make it hard for us to figure out which of those two conditions failed. I had such an example lately and we shouldn't do it again.